www-src/app/replicator/fromDevice/event_synchronizer.coffee

```diff
@@ -0,0 +1,125 @@
+async = require 'async'
+CozyToAndroidEvent = require "../transformer/cozy_to_android_event"
+AndroidCalendarCache = require "../cache/android_calendar_cache"
+AndroidCalendarHandler = require "../../lib/android_calendar_handler"
+log = require('../../lib/persistent_log')
+    prefix: "EventSynchronizer"
+    date: true
+continueOnError = require('../../lib/utils').continueOnError log
+
+module.exports = class EventSynchronizer
+
+    constructor: (@db, @calendarSync) ->
+        @db ?= app.replicator.config.db
+        @calendarSync ?= navigator.calendarsync
+        @androidCalendarHandler = new AndroidCalendarHandler()
+        @cozyToAndroidEvent = new CozyToAndroidEvent()
+        @androidCalendarCache = new AndroidCalendarCache()
```

Comment:
The actual cache is a "snapshot cache". It should be mentioned clearly (or upgraded to an uptodate cache).